calendar_base_booking/models/bookable_mixin.py

```diff
@@ -89,54 +81,73 @@ def _build_bookable_slot(self, open_slot, start, stop):
             start += relativedelta(minutes=delta)
         return bookable_slots
 
-    def get_open_slot(self, start, stop):
-        domain = self._get_domain(start, stop)
-        domain = expression.AND([domain, [("booking_type", "=", "bookable")]])
+    def get_bookable_slots(self, start, stop):
+        """
+        :return: all bookable slots
+        """
+        domain = expression.AND(
+            [
+                self._get_domain_slots_between(start, stop),
+                self._get_slot_domain_current_object(),
+                DOMAIN_BOOKABLE,
+            ]
+        )
         return self.env["calendar.event"].search(domain, order="start_date")
 
-    def get_bookable_slot(self, start, stop):
+    def get_available_slots(self, start, stop):
+        """
+        :return: bookable slots that are available (that are currently under capacity)
+        """
         start = fields.Datetime.to_datetime(start)
         stop = fields.Datetime.to_datetime(stop)
 
         slots = []
-        for open_slot in self.get_open_slot(start, stop):
-            for slot_start, slot_stop in self._get_available_slot(
-                max(open_slot.start, start), min(open_slot.stop, stop)
+        for bookable_slot in self.get_bookable_slots(start, stop):
+            for slot_start, slot_stop in self._get_available_slots(
+                max(bookable_slot.start, start), min(bookable_slot.stop, stop)
             ):
-                slots += self._build_bookable_slot(open_slot, slot_start, slot_stop)
+                slots += self._build_available_slots(
+                    bookable_slot, slot_start, slot_stop
+                )
         return slots
 
-    def _get_domain_for_current_object(self):
+    def get_booked_slots(self, start=None, stop=None):
+        if start and stop:
```

Comment:
I not sure that it's usefull to have the possibility to not pass a start and stop date ? (for the two other method it's required and I do not see case of use)